oneaccess.php

```diff
@@ -32,81 +33,59 @@
-		'admin_notices',
-		function () {
-			?>
-		<div class="notice notice-error">
-			<p>
-				<?php
-				printf(
-					/* translators: %s is the plugin name. */
-					esc_html__( 'You are running the %s plugin from the GitHub repository. Please build the assets and install composer dependencies to use the plugin.', 'oneaccess' ),
-					'<strong>' . esc_html__( 'OneAccess', 'oneaccess' ) . '</strong>'
-				);
-				?>
-			</p>
-			<p>
-				<?php
-				printf(
-					/* translators: %s is the command to run. */
-					esc_html__( 'Run the following commands in the plugin directory: %s', 'oneaccess' ),
-					'<code>composer install && npm install && npm run build:prod</code>'
-				);
-				?>
-			<p>
-				<?php
-				printf(
-					/* translators: %s is the plugin name. */
-					esc_html__( 'Please refer to the %s for more information.', 'oneaccess' ),
-					sprintf(
-						'<a href="%s" target="_blank">%s</a>',
-						esc_url( 'https://github.com/rtCamp/OneAccess' ),
-						esc_html__( 'OneAccess GitHub repository', 'oneaccess' )
-					)
-				);
-				?>
-			</p>
-		</div>
-			<?php
-		}
-	);
+	// load template-functions file to use oneaccess_get_template_content function.
+	require_once ONEACCESS_PLUGIN_LOADER_FEATURES_PATH . '/inc/helpers/template-functions.php';
+
+	echo oneaccess_get_template_content( 'notices/no-assets' ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped -- we are escaping the output in the template file.
 	return;
 }
 
-// phpcs:disable WordPressVIPMinimum.Files.IncludingFile.UsingCustomConstant
-if ( file_exists( ONEACCESS_PLUGIN_LOADER_FEATURES_PATH . '/vendor/autoload.php' ) ) {
-	require_once ONEACCESS_PLUGIN_LOADER_FEATURES_PATH . '/vendor/autoload.php';
-}
-// phpcs:enable WordPressVIPMinimum.Files.IncludingFile.UsingCustomConstant
+// Load the Composer autoloader.
+require_once ONEACCESS_PLUGIN_LOADER_FEATURES_PATH . '/vendor/autoload.php';
 
 /**
  * Load the plugin.
+ *
+ * @return void
  */
-function oneaccess_plugin_loader() {
+function oneaccess_plugin_loader(): void {
+
 	\OneAccess\Plugin::get_instance();
 
 	// load plugin text domain.
 	load_plugin_textdomain( 'oneaccess', false, ONEACCESS_PLUGIN_LOADER_RELATIVE_PATH . '/languages/' );
+
+	// if woocommerce action schedular is present into vendor then load it's file.
+	if ( file_exists( ONEACCESS_PLUGIN_LOADER_FEATURES_PATH . '/vendor/woocommerce/action-scheduler/action-scheduler.php' ) ) {
+		require_once ONEACCESS_PLUGIN_LOADER_FEATURES_PATH . '/vendor/woocommerce/action-scheduler/action-scheduler.php';
+	}
 }
 
-add_action( 'plugins_loaded', 'oneaccess_plugin_loader' );
+add_action( 'plugins_loaded', 'oneaccess_plugin_loader', -10 ); // added -10 to make sure action scheduler is loaded on 0 priority.
 
 /**
  * Activation hook to add roles.
  */
 register_activation_hook(
 	ONEACCESS_PLUGIN_LOADER_PLUGIN_BASENAME,
-	function () {
+	function (): void {
+
+		if ( ! class_exists( User_Roles::class ) && ! class_exists( DB::class ) ) {
+			return;
+		}
+
```

Comment:
always use FQCN in a class_exists check, but per the above comment, you should also FQCN to call the commands since we cant `use` in our root file.